### ISSUE TITLE
WebID

### ISSUE THREAD

**Issue by samuelgoto** [w3ctag/design-reviews#622]
Ya ya yawm TAG!

I'm requesting a TAG review of [WebID](https://github.com/WICG/WebID).

TL;DR; This is an **active** exploration to react to the **ongoing** privacy-oriented changes in browsers and **preserve** identity federation (e.g. OpenID, OAuth and SAML) on the web.

  - Explainer¹ (minimally containing user needs and example code): [url](https://github.com/WICG/WebID)
  - Security and Privacy self-review²: TODO(goto): will do. In the meantime, what we have [this](https://github.com/WICG/WebID/blob/main/privacy_threat_model.md) privacy threat model.
  - Primary contacts (and their relationship to the specification):
      - goto@chromium.org, http://github.com/samuelgoto, https://twitter.com/samuelgoto, Google
  - Organization/project driving the design: Google/Chrome
  - External status/issue trackers for this feature (publicly visible, e.g. Chrome Status): 
      - [WICG proposal](https://discourse.wicg.io/t/proposal-webid-privacy-preserving-federated-sign-in-api/4661)
      - [chromestatus](https://chromestatus.com/feature/6438627087220736)
      - [Intent To Prototype](https://groups.google.com/a/chromium.org/g/blink-dev/c/2B4TJ7j2U4M/m/1X5T3OszCAAJ)
      - [devtrial](https://github.com/WICG/WebID/blob/main/HOWTO.md)

Further details:

  - [x] I have reviewed the TAG's [Web Platform Design Principles](https://w3ctag.github.io/design-principles/)
  - The group where the incubation/design work on this is being done (or is intended to be done in the future): WICG
  - The group where standardization of this work is intended to be done ("unknown" if not known): unknown (WebAppSec seems the closest, OpenID foundation seems close too)
  - Existing major pieces of multi-stakeholder review or discussion of this design: many discussions are happening at many standards bodies, most notably the OpenID foundation and the OAuth WG
  - Major unresolved issues with or opposition to this design:
  - This work is being funded by: Google

You should also know that... 

- this is really early and we have a series of open questions.
- we are probably more interested in an evaluation / validation in exponentially decreasing interest:
   - First and foremost, did we get the **[problem](https://github.com/WICG/WebID#the-problem)** right?
       -  are we solving a real or a hypothetical problem? 
       - did we interpret the direction that browsers are going correctly? 
       - is there any precedence or comparable problem (beyond ads)? 
       - the tag is in a unique position to get a holistic perspective across browsers, where does it stand on this problem? 
   - Second, assuming that you agree with the problem statement and that identity federation is more secure compared to usernames/passwords, does the [end state](https://github.com/WICG/WebID#the-end-state) look directionally correct? 
   - Third, assuming that you agree with the problem and the direction that we are going is directionally correct, does the [sequence strategy](https://github.com/WICG/WebID#sequencing) make sense?
- if you are inclined to evaluate the solutions (rather than the problem), just wanted to provide some context: we haven't run into any easy solutions and most of the options come in the form of alternatives with trade-offs: a broad-but-shallow evaluation of the surface area (e.g.  an assessment of blind spots in multi-browser positions) is probably more effective to us now than than a narrow-but-deep evaluation of a specific formulation (e.g. API shape). In case you are lost in the many links, here are the specific APIs we are building to give a sense of what WebID looks like in practice:
    - [stage 1](https://github.com/WICG/WebID/blob/main/cookies.md) : Things we believe we need to preserve federation without 3p cookies
    - [stage 2](https://github.com/WICG/WebID/blob/main/navigations.md): Things we believe we'll need to preserve federation  under preventions against navigational tracking 
    - [stage 3](https://docs.google.com/presentation/d/1Sym0k84omyL5Ls1lO6w4aGQ-s4EHrDzo8ZlheyzFOlw/edit#slide=id.ga40b1e6d4f_0_77): Where we expect to park long term.
- if you meet over a VC and would welcome us joining, we are very happy to come and answer questions / clarify in real ti me

We'd prefer the TAG provide feedback as (please delete all but the desired option):

  💬 leave review feedback as a **comment in this issue** and @-notify @samuelgoto

**Comment by cynthia** [comment]
This name overlaps with existing work in W3C, which isn't ideal.

**Comment by samuelgoto** [comment]
> This name overlaps with existing work in W3C, which isn't ideal.

Ack, and on it. My mistake for missing the conflict. The very first line of the explainer contains the context, the status, our intent and how we are acting on it:

> not to be confused with [this](https://www.w3.org/2005/Incubator/webid/spec/) WebID whose authors have [graciously](https://github.com/WICG/WebID/issues/54#issuecomment-783605484) allowed us to use this as a codename until we [find](https://github.com/WICG/WebID/issues/41#issuecomment-712304910) a better one

**Comment by hadleybeeman** [comment]
Hi @samuelgoto! We are looking at this in our W3C TAG virtual face-to-face. 

Specifically looking at your [explainer](https://github.com/WICG/WebID) — we are having trouble working out what user need you are meeting (beyond "avoid the privacy pitfalls we've outlined"). For example: Are you looking to make it smoother  for a user to log in with a federated ID? Are you looking to display on the page "Hello Hadley, would you like to sign in with your Google* account?" (Other ID providers obviously are available.) Are you keen to make social buttons work better so the user can post/share a link to the site they're on?

It would be great if you could help us understand what user needs you are focusing on. (It would also be great if you could put them in the explainer too.) Thanks!

**Comment by hadleybeeman** [comment]
Hi @samuelgoto — we are revisiting this in our break-out session. We understand that you are considering creating a community group around this? We are very supportive, and are keen to see this space investigated further. Please do keep us in the loop as you develop designs for new web features — we'd love to review and help with them.

**Comment by samuelgoto** [comment]
> Hi @samuelgoto — we are revisiting this in our break-out session. We understand that you are considering creating a community group around this? 

Yes. [Here](https://github.com/hlflanagan/fedidcg/blob/gh-pages/Federated%20Identity%20Community%20Group%20Charter.md) is how this is staring to form.

> We are very supportive, and are keen to see this space investigated further. Please do keep us in the loop as you develop designs for new web features — we'd love to review and help with them.

Will do.

> In light of that, we propose closing this issue.

SGTM!

Thanks for the early review, much appreciated! We will kick off with more in-depth and specific full reviews for parts of the API surface!

Thanks all, Sam
